invoice_discount_display_amount/models/account_move.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2022 Manuel Regidor <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+
+    _inherit = "account.move"
+
+    discount_total = fields.Monetary(
+        compute="_compute_discount_total",
+        string="Discount Total",
+        currency_field="currency_id",
+        store=True,
+    )
+    price_total_no_discount = fields.Monetary(
+        compute="_compute_discount_total",
+        string="Total Without Discount",
+        currency_field="currency_id",
+        store=True,
+    )
+
+    @api.depends(
+        "invoice_line_ids.discount_total", "invoice_line_ids.price_total_no_discount"
+    )
+    def _compute_discount_total(self):
+        for invoice in self:
```

Comment:
you should filter those account.move that are not invoices and directly indicate that the value of the fields is 0.0 with a massive change.